src/pkg/packaging-tools/framework.dependency.targets

```diff
@@ -90,6 +90,243 @@
+      <_crossDir Condition="'$(TargetArchitecture)' == 'arm64'">/x64_arm64</_crossDir>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'@(_runtimeCLR)' != ''">
+      <_crossGenPath>$(_runtimePackageDir)tools$(_crossDir)/crossgen$(ApplicationFileExtension)</_crossGenPath>
+      <_runtimeDirectory>%(_runtimeCLR.RootDir)%(_runtimeCLR.Directory)</_runtimeDirectory>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'@(_runtimeCoreLib)' != ''">
+      <_coreLibDirectory>%(_runtimeCoreLib.RootDir)%(_runtimeCoreLib.Directory)</_coreLibDirectory>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'@(_runtimeJIT)' != ''">
+      <_jitPath>%(_runtimeJIT.FullPath)</_jitPath>
+      <_jitPath Condition="'$(_crossDir)' != ''">$(_jitPackageDir)runtimes$(_crossDir)/native/$(LibraryFilePrefix)clrjit$(LibraryFileExtension)</_jitPath>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'@(_fxSystemRuntime)' != ''">
+      <_fxLibDirectory>%(_fxSystemRuntime.RootDir)%(_fxSystemRuntime.Directory)</_fxLibDirectory>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'@(_windowsWinMD)' != ''">
+      <_windowsWinMDDirectory>%(_windowsWinMD.RootDir)%(_windowsWinMD.Directory)</_windowsWinMDDirectory>
+    </PropertyGroup>
+    
+    <ItemGroup>
+      <_requiredProperty Include="_runtimePackageDir;_coreLibDirectory;_crossGenPath;_jitPath;_fxLibDirectory;_windowsWinMDDirectory" />
+    </ItemGroup>
+
+    <Message Text="%(_requiredProperty.Identity): $(%(_requiredProperty.Identity))" />
+    <Error Condition="'$(%(_requiredProperty.Identity))' == ''" Text="Required property %(_requiredProperty.Identity) was not set." />
+    <Error Condition="!Exists('$(%(_requiredProperty.Identity))')" Text="Required property %(_requiredProperty.Identity) with value '$(%(_requiredProperty.Identity))' does not exist." />
+  </Target>
+
+  <!-- Prepares all items for cross-gen and replaces package file items with their cross-gen'ed equivalents -->
+  <Target Name="PrepareForCrossGen"
+          Condition="'$(NuGetRuntimeIdentifier)' != '' AND '$(DisableCrossgen)' != 'true'"
+          DependsOnTargets="
+            GetCrossgenPackagePaths;
+            GetFilesFromPackages">
+    <PropertyGroup>
+      <_crossGenIntermediatePath>$(IntermediateOutputPath)/crossgen</_crossGenIntermediatePath>
+    </PropertyGroup>
+    <!-- Ensure crossgen is executable.  See https://github.com/NuGet/Home/issues/4424 -->
+    <Exec Command="chmod u+x $(_crossGenPath)"
+          Condition="'$(OSGroup)' != 'Windows_NT'" />
+    <ItemGroup>
+      <!-- Exclude run crossgen on System.Runtime.WindowsRuntime for non-Windows due to System.Runtime.WindowsRuntime.dll for non-Windows only throw NotSupported Exception  -->
+      <!-- Also crossgen on non-windows doesn't support -Platform_Winmd_Paths -->
+      <!-- Exclude crossgen for resource DLLs in 'cs/', 'de/', ... subdirectories. -->
```

Comment:
I'm actually not sure why I assume resource DLLs shouldn't be crossgenned. Let me know if these should be, or if there are concrete reasons either way. I'll merge this PR without crossgenning them if I don't hear otherwise.

Reply:
isn't crossgening resource dll's sort of a no-op? Because they contain no logical code?

Reply:
That would make sense, but crossgenning reduces the size of e.g. `de/PresentationCore.resources.dll` from 103 KB to 94 KB. 😕

Reply:
/cc @fadimounir

Reply:
GitHub won't let me attach the input/output to help compare, but this RSP will do it in the repro dir I shared via email:

```
-readytorun
-in wd\de\PresentationCore.resources.dll
-out wd-out/PresentationCore.resources.dll
-platform_assemblies_paths wd\;corefx\;coreclr\
-Platform_Winmd_Paths winmd
-JITPath jit\clrjit.dll
```

Reply:
Actually, filed https://github.com/dotnet/core-setup/issues/5975 to track this better rather than a PR comment thread.